### DISCUSSION TITLE
Engaging community

### DISCUSSION THREAD

**Discussion by taylorific** [polymathrobotics/caladan_examples#8] [Ideas & Requests]
@nroccaforte Do you have some thoughts/ideas right now on building community in this discussion forum? I received feedback from @iliabaranov that having the code pipelines format and lint code was a bit of a downer. It is possible for us to hook nearly any activity in and around a github repo. Perhaps we might want to send automated emails/messages that send congratulations and positive reinforcement when they use the formus, submit issues, pull request, etc. Your thoughts?

**Comment by iliabaranov** [comment]
Slight correction, I was wondering if this could be a barrier to entry, not sure if it is or isn't yet (testing a simple setup process this week, if it's easy for people to do, should be no barrier :) )

**Comment by amazing-bettyanne** [comment]
Yeah, I'm not sure. It's easy enough to run the formatter locally from our images with the following container image we've posted on dockerhub, like so, it's not exactly obvious. And ros itself doesn't seem to have any automatic CI things happening in their core stacks, so this might come as a surprise to roboticists: https://github.com/ros

```
docker run --rm \
  --mount type=bind,source="$(pwd)",target=/code \
  polymathrobotics/black .
```

The idea for this pipeline came mostly from an internal issue where we had a few code repos where different team members had radically different ways of formatting their code, to the extent that it was almost impossible to figure out if code submitted by one person had been changed by another because of all the formatting and reformatting going on. Caused utter chaos there until we started adding an automated formatting check as part of the pull request process. This at least made it so that people submitted code that looked mostly the same.

Is this needed for our API examples? Not sure.

Also the choice of black as a code formatter might be controversial: https://github.com/psf/black It's very opinionated and doesn't allow a lot of configuration besides line length. Does the formatting that black recommends hurt your eyes too much? Is that a concern?

**Comment by taylorific** [comment]
Hey @amazing-bettyanne can you give me rights to try a few new hooks in the repo to try a few things around triggering messages on events? I'd like to test some of the possibilities.

**Comment by amazing-bettyanne** [comment]
Sure @taylorific I've given you rights so you can push things to the GitHub Actions workflows in the repo - amaze and amuse me!

**Reply by taylorific** [reply]
Thanks! I think I have performance anxiety now! I'll try a few things that might not be too terrible - I live to underwhelm....

**Comment by nroccaforte** [comment]
I love the idea of notifications based on engagement!

I think it’d help for our (Polymath) team to post known bugs/issues here to start, and the resolutions as we find them.

Could also be fun to do a weekly theme post, like Friday robot memes or Dune content. Let's see what people think :)
